### DISCUSSION TITLE
2 problems in using dpgen

### DISCUSSION THREAD

**Discussion by imindshejian** [deepmodeling/dpgen#366] [Q&A] [answered]
1. There are 5000 frames data in the initset/set.00[0-4],each of them have 1000 frames, an error message were reported during iter.000000 task 01: **RuntimeError: do not find download file /home/kemove/hejian/dp_work/gen/18a0d563-3b7f-4464-b182-774c2e8cbf5a/000/frozen_model.pb**
then i found error in train.log:
Traceback (most recent call last):
  **File "/home/kemove/deepmd-kit/bin/dp", line 10, in <module>
    sys.exit(main())
  File "/home/kemove/deepmd-kit/lib/python3.8/site-packages/deepmd/main.py", line 73, in main
    train(args)
  File "/home/kemove/deepmd-kit/lib/python3.8/site-packages/deepmd/train.py", line 87, in train
    _do_work(jdata, run_opt)
  File "/home/kemove/deepmd-kit/lib/python3.8/site-packages/deepmd/train.py", line 128, in _do_work
    data = DeepmdDataSystem(systems, 
  File "/home/kemove/deepmd-kit/lib/python3.8/site-packages/deepmd/DataSystem.py", line 27, in __init__
    self.data_systems.append(DeepmdData(ii, 
  File "/home/kemove/deepmd-kit/lib/python3.8/site-packages/deepmd/Data.py", line 37, in __init__
    self.test_dir = self.dirs[-1]
IndexError: list index out of range**
it seems a variable of test sets wrong, but adjusting "numb_test" doesn't work, what should i change in param.json.

2. I've try to train 4 models by using deepmd-kit mannually, and copy the result file to the path that dpgen generated, the dpgen workflow successfully go to task06, the model_devi includes 10 configs, some of MD were failed possibly due to the initial config were too bad, the model_devi.log reported that:
**Invalid argument: 2 root error(s) found.
  (0) Invalid argument: Assert failed, max neighbor size of atom(lammps) 20471 is larger than 4096, which currently is not supported by deepmd-kit.
	 [[{{node DescrptSeA}}]]
  (1) Invalid argument: Assert failed, max neighbor size of atom(lammps) 20471 is larger than 4096, which currently is not supported by deepmd-kit.
	 [[{{node DescrptSeA}}]]
	 [[o_energy/_25]]
0 successful operations.
0 derived errors ignored.**
and no traj were generated, but some of configs were successfully done. Do every MD task in model_devi should have a good initial configs to ensure the MD run succesfully? If the failure of MD make the dpgen workflow stopped? Because there are no message about how many configs were selected to next fp calculation in dpgen.log.

**Comment by AnguseZhang** [comment, marked as answer]
Problem 1: 
Seems like the number of frames in  some of your systems  is less than `numb_test` settings in your `param.json`. Since you have manually modified `input.json` and successfully finished the training, you can just do the same modifications on training parameters in `param.json`, and restart from a proper stage via the file `record.dpgen`

Problem 2:
The maximal neighbors of an atom in DeePMD-kit is up to 4096, as the log said. "20471"  is quite rare for MD simulations. You should carefully check your initial configurations and trained models.  Currently the MD run should finish successfully, you can use small total steps of MD in early iterations where the quality of model is not quite so robust.
